gradle-plugin/openshift/src/test/java/org/eclipse/jkube/gradle/plugin/task/TaskEnvironmentExtension.java

```diff
@@ -13,46 +13,48 @@
  */
 package org.eclipse.jkube.gradle.plugin.task;
 
-import java.io.File;
-import java.io.IOException;
-import java.util.ArrayList;
-import java.util.List;
-import java.util.stream.Stream;
-
-import org.eclipse.jgit.util.FileUtils;
 import org.gradle.api.DefaultTask;
 import org.gradle.api.Project;
 import org.gradle.api.artifacts.Configuration;
 import org.gradle.api.artifacts.ConfigurationContainer;
 import org.gradle.api.internal.plugins.DefaultPluginContainer;
 import org.gradle.api.logging.Logger;
 import org.gradle.api.plugins.JavaPluginConvention;
-import org.junit.rules.TemporaryFolder;
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
 import org.mockito.MockedConstruction;
 
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Stream;
+
 import static org.mockito.Mockito.RETURNS_DEEP_STUBS;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.mockConstruction;
 import static org.mockito.Mockito.when;
 
-public class TaskEnvironment extends TemporaryFolder {
-
-  private MockedConstruction<DefaultTask> defaultTaskMockedConstruction;
-  Logger logger;
+public class TaskEnvironmentExtension implements BeforeEachCallback, AfterEachCallback {
   Project project;
+  Logger logger;
+  File temporaryFolder;
```

Comment:
Maybe keep this one private and provide a getter `getRoot` so there's no need for the extra-changes in the other extension-dependent test classes.